src/libstd/keyword_docs.rs

```diff
@@ -925,8 +925,81 @@ mod mut_keyword {}
 //
 /// Make an item visible to others.
 ///
-/// The documentation for this keyword is [not yet complete]. Pull requests welcome!
+/// The keyword `pub` makes any module, function, or data structure accessible from inside
+/// of external modules. The `pub` keyword may also be used in a `use` declaration to re-export
+/// an identifier from a namespace.
+///
+///  In the below example we define `PublicStruct` within `public_inner_module` and make both public
```

Comment:
In thinking about this some more, I would personally prefer to drop the code examples from this documentation and primarily lead people towards the [visibility documentation in the reference](https://doc.rust-lang.org/reference/visibility-and-privacy.html).

I think in practice that means leaving the first paragraph in, and then deleting the code examples and such up to the final paragraph. I think that this is just not the best place for long-form documentation like this.